.pre-commit-config.yaml

```diff
@@ -2,18 +2,26 @@ repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v2.3.0
     hooks:
-    -   id: check-yaml
-    -   id: end-of-file-fixer
-    -   id: trailing-whitespace
-    -   id: flake8
-  - repo: https://github.com/ambv/black
-    rev: stable
+    - id: check-yaml
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
+    - id: flake8
+      args: [--max-line-length=131]
+
+  - repo: https://github.com/psf/black
+    rev: 19.10b0 # Replace by any tag/version: https://github.com/psf/black/tags
     hooks:
     - id: black
       language_version: python3.8
+      args: ["--line-length=120"]
 
   - repo: https://github.com/pycqa/isort
     rev: 5.5.2
     hooks:
-      - id: isort
-        files: 'emmet-core/*'
+    - id: isort
+      files: 'emmet-core/.*'
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.790'
+    hooks:
+      - id: mypy
```

Comment:
I've always found this to be slow. It's also hard to enforce on a commit-by-commit basis.

Reply:
OK removed, I do like it in general though is there a way to locally turn it on?